unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/LoadingScreenScripts/LoadingScreenHintsController.cs

```diff
@@ -41,14 +65,14 @@ public LoadingScreenHintsController(HintRequestService hintRequestService)
         private async void InitializeHintsAsync()
         {
             cancellationTokenSource = new CancellationTokenSource();
-
-            IAddressableResourceProvider  addressableProvider = new AddressableResourceProvider();
-            hintViewPrefab = await addressableProvider.GetAddressable<HintView>(SOURCE_HINT_ADDRESSABLE, cancellationTokenSource.Token);
+            hintViewPrefab = await addressableProvider.GetAddressable<HintView>(HINT_VIEW_PREFAB_ADDRESSABLE, cancellationTokenSource.Token);
+            var hintsContainer = loadingScreenView.GetHintContainer();
 
             // Initializing empty hints views
-            for (int i = 0; i < MAX_HINTS; i++)
+            for (var i = 0; i < MAX_HINTS; i++)
             {
-                HintView newHintView = Object.Instantiate(hintViewPrefab);
+                HintView newHintView = Object.Instantiate(hintViewPrefab, hintsContainer, false);
+                newHintView.transform.localPosition = Vector3.zero;
                 newHintView.ToggleHint(false);
                 hintViewPool.Add(newHintView);
             }
```

Comment:
There are a few problems here:

 It breaks a bit our MVC conventions and also our best practices for dealing with Monobehaviours. The controller should never have dependencies towards Unity objects (such as a container). Those are implementation details for the view and should be handled there.


It also breaks the Dependency Inversion and the Interface Segregation principles.
I dont agree that much with having the `HintView` handling in this controller, I think that's a loadingScreenView implementation detail (as an example, view could have a virtual list, a single one being redrawn... and that wouldnt affect the controller). But nonetheless I dont find the approach conceptually wrong.

The problem here is the lack of a public interface to handle HintViews. In fact I have no idea how is this being tested, because you cannot mock `HintView`. Is the test suite for LoadingScreenHintsController actually instantiating HintViews?

At minimum I'd extract a `IHintView` interface, have a factory and put there all the pooling logic + handling of references dependencies. In the end `LoadingScreenHintsController` should just receive a provider/factory of `IHintViews`